### PR TITLE
Render juicer feed

### PR DESCRIPTION
**Pull request by joeleenk**
Mostly front end code:

- Output data in templates
- Add css + js for styles and accessibility
- AJAX for load more
- Add some php functions and logic to support custom read more links

### CODE REVIEW

**Comment by rmccue** [comment]
@joeleenk Looks like you've committed the `vendor` directory here which is installed via Composer; that should be `.gitignore`'d :)

File: inc/api.php [hunk 1]
```diff
@@ -89,6 +89,87 @@ function prepare_post_items( array $items ) : array {
...
+		$read_more_src = $social_post_src;
+
+	} else {
+		$post_type     = 'post';
+		$read_more_src = $external_src_href;
+	}
+
+	// Set text based on post type.
+	if ( 'photo' === $post_type ) {
+		// translators: the item source (Facebook, Twitter, Original Website).
```
Comment (jazzsequence) [inline inc/api.php]: Translator comments should be capitalized (like any other comment) and should note what the comment is referring to (in this case the `%s`. The idea here is that a translator will be using a program to translate strings and might need context to understand what the variable in the string is referring to to provide an accurate translation.
🔢 

```suggestion
		// Translators: %s refers to the item source (Facebook, Twitter, Original Website).
```

File: inc/api.php [hunk 2]
```diff
@@ -98,13 +179,30 @@ function prepare_post_items( array $items ) : array {
...
+	// Get message from post.
 	$content = wp_kses( make_clickable( $message ), allowed_html() );
+	// Search content for links.
 	preg_match( '/<a ?.*>(.*)<\/a>/', $content, $link_matches );
 
-	if ( $item->external === $link_matches[1] ) {
-		$link_text = __( 'Read More', 'hm-juicer' );
-		$link_url  = $link_matches[1];
-		$content   = str_replace( "<a href=\"$link_url\">$link_url</a>", "<a href=\"$link_url\" class=\"juicer-post__sharing-link\">$link_text</a>", $content );
+	$full_read_more_link = "<a href=\"$link_url\" class=\"juicer-post__sharing-link\" aria-label=\"$link_text, posted $posted_time on $social_post_src\">$link_text <i class=\"fas fa-chevron-right\" aria-hidden=\"true\"></i></a>";
```
Comment (jazzsequence) [inline inc/api.php]: Aria label should be i18nized

```suggestion
	// Translators: 1: The link text, 2: the time the link was posted, 3: the social network the link was posted to.
	$aria_label = sprintf( __( '%1$s, posted %2ds on %3$s', 'hm-juicer' ), $link_text, $posted_time, $social_post_src );
	$full_read_more_link = "<a href=\"$link_url\" class=\"juicer-post__sharing-link\" aria-label=\"$aria_label">$link_text <i class=\"fas fa-chevron-right\" aria-hidden=\"true\"></i></a>";
```

File: inc/functions.php [hunk 3]
```diff
@@ -12,11 +12,15 @@
...
  * Display the main Juicer social media feed.
  *
- * @param int $count The number of posts to display.
- * @param int $page  The page to display.
+ * @param int  $count     The number of posts to display.
+ * @param int  $page      The page to display.
+ * @param bool $load_more Whether to include a load more button.
  */
-function juicer_feed( $count = 10, $page = 1 ) {
+function juicer_feed( $count = 10, $page = 1, $load_more = false ) {
```
Comment (jazzsequence) [inline inc/functions.php]: We can typecast these parameters here, too.

```suggestion
function juicer_feed( int $count = 10, int $page = 1, bool $load_more = false ) {
```

File: inc/load-more.php [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+	return $response;
+}
+
+/**
+ * Enqueue styles and scripts.
+ */
+function enqueue_scripts() {
+	// Enqueue custom JS for the HM Juicer layout.
+	Asset_Loader\register_script( [
```
Comment (jazzsequence) [inline inc/load-more.php]: Note: we'll need to include the HM Asset Loader as part of this new repo

File: assets/juicer.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
+/* global imagesLoaded */
```
Comment (jazzsequence) [inline assets/juicer.js]: It would be good if the js files were in a `/js` subfolder rather than in the root `assets` folder

Comment (joeleenk) [reply assets/juicer.js]: Yep, I saw you changed it in the config file. Fixed in [b15cbac](https://github.com/humanmade/hm-juicer/pull/6/commits/b15cbacefb68689ae8c83f17b884be0cadecaf57)

File: assets/scss/_cards.scss [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+ */
+
+.juicer-grid {
+	display: flex;
+	flex-flow: row wrap;
+
+	@supports (display: grid) {
+		display: grid;
+		grid-gap: 1em;
+		grid-template-columns: repeat(auto-fill, minmax(350px,1fr));
```
[the remark this answers is not in the document]
Comment (joeleenk) [reply assets/scss/_cards.scss]: Fixed in [fe5683a](https://github.com/humanmade/hm-juicer/pull/6/commits/fe5683a216f8e77a61cd209d7c6853990259608f)

File: assets/scss/_cards.scss [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+		transition: opacity 1s ease-in-out;
+
+		@media screen and ( min-width: $bp-m-s ) {
+			flex: 0 0 48.5%;
+
+			&:nth-child( 2n+1 ) {
+				margin-right: 3%;
+
+				@supports (display: grid) {
+					margin-right: 0;
```
[the remark this answers is not in the document]
Comment (joeleenk) [reply assets/scss/_cards.scss]: Fixed in [fe5683a](https://github.com/humanmade/hm-juicer/pull/6/commits/fe5683a216f8e77a61cd209d7c6853990259608f)

File: assets/scss/_cards.scss [hunk 8]
```diff
@@ -0,0 +1,295 @@
+/**
+ * Styles for HM Juicer masonry layout.
+ */
+
+.juicer-grid {
+	display: flex;
+	flex-flow: row wrap;
+
+	@supports (display: grid) {
```
[the remark this answers is not in the document]
Comment (joeleenk) [reply assets/scss/_cards.scss]: Fixed in [fe5683a](https://github.com/humanmade/hm-juicer/pull/6/commits/fe5683a216f8e77a61cd209d7c6853990259608f)

File: inc/api.php [hunk 9]
```diff
@@ -89,6 +89,87 @@ function prepare_post_items( array $items ) : array {
...
+/**
+ * Get the read more link text, based on the post type and source.
+ *
+ * @param string $external_src_link   The URL of the original post source (photo, video, linked article).
+ * @param string $social_post_src     The social network the post is pulled from.
+ * @param string $social_profile_name The profile name of the social network account.
+ *
+ * @return string                     The filtered Juicer post message.
+ */
+function get_read_more_text( $external_src_link, $social_post_src, $social_profile_name ) {
```
[the remark this answers is not in the document]
Comment (joeleenk) [reply inc/api.php]: Fixed in [e265b3b](https://github.com/humanmade/hm-juicer/pull/6/commits/e265b3bb7c03ddea9243b79b7fac536dccd671d3)

File: inc/functions.php (hunk 3 above)
Comment (joeleenk) [reply inc/functions.php]: Fixed in [e265b3b](https://github.com/humanmade/hm-juicer/pull/6/commits/e265b3bb7c03ddea9243b79b7fac536dccd671d3)

File: inc/api.php (hunk 2 above)
Comment (joeleenk) [reply inc/api.php]: Fixed in [1021254](https://github.com/humanmade/hm-juicer/pull/6/commits/1021254e6a720b596ee55f1fdb7b0dde2f580ca7)

File: inc/api.php (hunk 1 above)
Comment (joeleenk) [reply inc/api.php]: Fixed in [1021254](https://github.com/humanmade/hm-juicer/pull/6/commits/1021254e6a720b596ee55f1fdb7b0dde2f580ca7)

File: assets/load-more.js [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+			$loading.removeClass( 'loading' );
+		} );
+	}, 300 );
+
+	$button.on( 'click', function ( e ) {
+		e.preventDefault();
+		loadMorePosts();
+	} );
+
+} )( window, jQuery );
```
[the remark this answers is not in the document]
Comment (joeleenk) [reply assets/load-more.js]: Fixed in [8ed47e6](https://github.com/humanmade/hm-juicer/pull/6/commits/8ed47e62b125fb89c81ba36bb256e85ba02fa03c)

File: assets/scss/_cards.scss [hunk 11]
```diff
@@ -42,24 +31,31 @@
...
 
+@supports ( display: grid ) {
+	.juicer-grid {
+		display: grid;
+		grid-gap: 1em;
+		grid-template-columns: repeat( auto-fill, minmax( 350px,1fr ) );
+		grid-auto-rows: 0;
+
+		// Be specific here to override the IE11 fallbacks.
+		& &__item {
```
Comment (jazzsequence) [inline assets/scss/_cards.scss]: The comment here doesn't seem to align with the code. IE11 doesn't support `display: grid` so this wouldn't work as an IE11 fallback.

Comment (joeleenk) [reply assets/scss/_cards.scss]: The comment says we're **overriding** the IE11 fallback. I can make it more specific though, if that would help.

**Pull request merged**